data_steward/cleaners/clean_cdr.py

```diff
@@ -1,145 +1,171 @@
-import oauth2client
 
 # Project imports
 import bq_utils
-#import constants.bq_utils as bq_consts
-import constants.cleaners.combined as combined_consts
-import constants.cleaners.combined_deid as deid_consts
-import constants.cleaners.ehr as ehr_consts
-import constants.cleaners.rdr as rdr_consts
-import constants.cleaners.unioned as unioned_consts
+import clean_cdr_engine as clean_engine
+import constants.cleaners.clean_cdr as clean_cdr_consts
+import id_deduplicate as id_dedup
 
-LOGGER = logging.getLogger(__name__)
-
-def _add_console_logging(add_handler):
-    # this config should be done in a separate module, but that can wait
-    # until later.  Useful for debugging.
-    logging.basicConfig(level=logging.DEBUG,
-                        filename='/tmp/cleaner.log',
-                        filemode='a')
-
-    if add_handler:
-        handler = logging.StreamHandler()
-        handler.setLevel(logging.INFO)
-        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-        handler.setFormatter(formatter)
-        LOGGER.addHandler(handler)
-
-
-def _clean_dataset(project=None, dataset=None, statements=None):
-    if project is None or project == '' or project.isspace():
-        project = app_identity.get_application_id()
-        LOGGER.debug('Project name not provided.  Using default.')
-
-
-    if statements is None:
-        statements = []
-
-    failures = 0
-    successes = 0
-    for statement in statements:
-        full_query = statement.format(project=project, dataset=dataset)
-
-        try:
-            results = bq_utils.query(full_query)
-        except (oauth2client.client.HttpAccessTokenRefreshError,
-            googleapiclient.errors.HttpError):
-            LOGGER.exception("FAILED:  Clean rule not executed:\n%s", full_query)
-            failures += 1
-            continue
-
-        LOGGER.info("Executing query %s", full_query)
-
-        # wait for job to finish
-        query_job_id = results['jobReference']['jobId']
-        incomplete_jobs = bq_utils.wait_on_jobs([query_job_id])
-        if incomplete_jobs != []:
-            failures += 1
-            raise bq_utils.BigQueryJobWaitError(incomplete_jobs)
+# import constants.bq_utils as bq_consts
 
-        successes += 1
+LOGGER = logging.getLogger(__name__)
 
-    if successes > 0:
-        LOGGER.info("Successfully applied %d clean rules for %s.%s",
-                    successes, project, dataset)
-    else:
-        LOGGER.warning("No clean rules successfully applied to %s.%s",
-                       project, dataset)
 
-    if failures > 0:
-        print("Failed to apply %d clean rules for %s.%s",
-                       failures, project, dataset)
-        LOGGER.warning("Failed to apply %d clean rules for %s.%s",
-                       failures, project, dataset)
+def _gather_ehr_queries(project, dataset):
+    """
+    gathers all the queries required to clean ehr dataset
```

Comment:
This is the only new comment.  When writing docstrings, write a single line summary statement, followed by a blank line.  You may then add more descriptions and parameter definitions.  Just separate sections with blank lines.  This is from [pep-257](https://www.python.org/dev/peps/pep-0257/).  [Pep-8 docstrings documentation](https://www.python.org/dev/peps/pep-0008/#documentation-strings) references pep-257.